### PR TITLE
send request to add to cart contrllor

### PR DESCRIPTION
**Pull request by momenmarwan**

### CODE REVIEW

**Review by Farah202324 — APPROVED** [review]

File: client/src/components/Cart/DownComponet.tsx [hunk 1]
```diff
@@ -1,8 +1,53 @@
 import * as core from '@mui/material';
+import axios from 'axios';
+import { FC, useEffect, useState } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:4]: remove this comment.

File: client/src/components/Cart/DownComponet.tsx [hunk 2]
```diff
@@ -1,8 +1,53 @@
...
+import { useMutation } from 'react-query';
+import { useNavigate } from 'react-router-dom';
+import { ButtonSectionProps } from '../../utils';
 
-const { Button } = core;
+const { Button, Alert } = core;
+
+const serverUrl = import.meta.env.VITE_APP_SERVER_URL;
+
+const DownComponent: FC<ButtonSectionProps> = ({ notes, setNotes }) => {
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:13]: remove setNotes since it is not used.

File: client/src/components/Cart/DownComponet.tsx [hunk 3]
```diff
@@ -1,8 +1,53 @@
...
+  const goToWaitingRoom = useNavigate();
+
+  const [orders, setOrders] = useState<[] | null>([]);
+  const { isError, isSuccess, isLoading, mutate } = useMutation(
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:17]: remove isLoading since it is not used.

File: client/src/components/Cart/DownComponet.tsx [hunk 4]
```diff
@@ -1,8 +1,53 @@
...
+    'post',
+    (reqBody: any) => {
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:19]: Do not use any!

File: client/src/components/Cart/DownComponet.tsx [hunk 5]
```diff
@@ -1,8 +1,53 @@
...
+    },
+  );
+  const handleCheckout = () => {
+    const body = {
+      orders,
+      note: notes,
+      customerId: 123456,
+    };
+    mutate(body);
+    console.log(isError);
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:30]: Remove the logs.

File: client/src/components/Cart/DownComponet.tsx [hunk 6]
```diff
@@ -1,8 +1,53 @@
...
+    console.log(isSuccess);
+  };
+
+  useEffect(() => {
+    let dataFromLocalStorage = JSON.parse(localStorage.getItem('items') as string);
+    dataFromLocalStorage = dataFromLocalStorage.map((dish: { id: any; count: any }) => {
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:36]: Do not use any!

File: client/src/components/Cart/Pop.tsx [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { useState, FC } from 'react';
 import InsertCommentIcon from '@mui/icons-material/InsertComment';
 import { ButtonSectionProps } from '../../utils';
 
-const { TextField, Button } = core;
+const { TextField, Button, Typography } = core;
 const ButtonSection: FC<ButtonSectionProps> = ({ notes, setNotes }) => {
```
Comment (moustf) [inline client/src/components/Cart/Pop.tsx:12]: Remove notes since it is not used.

File: client/src/pages/Cart.tsx [hunk 8]
```diff
@@ -1,14 +1,22 @@
+import { useState } from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (moustf) [inline client/src/pages/Cart.tsx:2]: Remove this log. why did you add it?

File: client/src/pages/Cart.tsx [hunk 9]
```diff
@@ -1,14 +1,22 @@
...
+import { QueryClient, QueryClientProvider } from 'react-query';
 import CartNavBar from '../components/Cart/CartNavBar';
 import CartList from '../components/Cart/CartList';
 import DownComponent from '../components/Cart/DownComponet';
 
+const queryClient = new QueryClient();
```
Comment (moustf) [inline client/src/pages/Cart.tsx:8]: Why did you add a new query client class? Farah initialized one in the main.tsx and added its provider to all the components.

File: client/src/pages/Cart.tsx (hunk 8 above)
Comment (moustf) [inline client/src/pages/Cart.tsx:2]: Remove this log. why did you add it?

File: client/src/pages/Cart.tsx (hunk 9 above)
Comment (moustf) [inline client/src/pages/Cart.tsx:8]: Why did you add a new query client class? Farah initialized one in the main.tsx and added its provider to all the components.

File: client/src/pages/Cart.tsx [hunk 10]
```diff
@@ -1,14 +1,22 @@
...
+
 const CartPage = () => {
+  const [notes, setNotes] = useState<string>('');
   return (
-    <div>
-      <CartNavBar />
-      <CartList />
-      <DownComponent />
-    </div>
+    <QueryClientProvider client={queryClient}>
```
Comment (moustf) [inline client/src/pages/Cart.tsx:13]: Remove the query provider!

File: client/src/components/Cart/Pop.tsx (hunk 7 above)
Comment (moustf) [inline client/src/components/Cart/Pop.tsx:12]: Remove notes since it is not used.

File: client/src/components/Cart/DownComponet.tsx (hunk 1 above)
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:4]: Remove this comment.

File: client/src/components/Cart/DownComponet.tsx (hunk 2 above)
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:13]: Remove setNotes since it is not used.

File: client/src/components/Cart/DownComponet.tsx (hunk 3 above)
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:17]: Remove isLoading since it is not used.

File: client/src/components/Cart/DownComponet.tsx (hunk 4 above)
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:19]: Do not use any!

File: client/src/components/Cart/DownComponet.tsx [hunk 11]
```diff
@@ -1,8 +1,53 @@
...
+  );
+  const handleCheckout = () => {
+    const body = {
+      orders,
+      note: notes,
+      customerId: 123456,
+    };
+    mutate(body);
+    console.log(isError);
+    console.log(isSuccess);
```
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:31]: Remove the logs.

File: client/src/components/Cart/DownComponet.tsx (hunk 6 above)
Comment (moustf) [inline client/src/components/Cart/DownComponet.tsx:36]: Do not use any!

**Review by moustf — CHANGES_REQUESTED** [review]

**Review by momenqudaih — APPROVED** [review]

**Review by moustf — APPROVED** [review]
